test/chat_api/conversations_test.exs

```diff
@@ -41,11 +41,39 @@ defmodule ChatApi.ConversationsTest do
       different_customer = customer_fixture(different_account)
       _conversation = conversation_fixture(different_account, different_customer)
 
+      for _n <- 1..10,
+          do: message_fixture(account, conversation, %{sent_at: ~N[2020-11-14 20:50:26]})
+
       result_ids = Enum.map(Conversations.list_conversations_by_account(account.id), & &1.id)
 
       assert result_ids == [conversation.id]
     end
 
+    test "list_conversations_by_account_v2/2 returns all conversations with one message(most recent)",
+         %{
+           account: account,
+           conversation: conversation
+         } do
+      different_account = account_fixture()
+      different_customer = customer_fixture(different_account)
+      conversation_fixture(different_account, different_customer)
+      secondary_conversation = conversation_fixture(account, different_customer)
+
+      for _n <- 1..5,
+          do: message_fixture(account, conversation, %{sent_at: ~N[2020-11-14 20:50:26]})
+
+      for _n <- 1..3,
+          do:
+            message_fixture(account, secondary_conversation, %{sent_at: ~N[2020-11-14 20:50:26]})
+
+      conversations = Conversations.list_conversations_by_account_v2(account.id, %{})
+      assert length(conversations) == 2
+
+      assert [%{messages: messages}, %{messages: messages2}] = conversations
+      assert length(messages) == 1
+      assert length(messages2) == 1
```

Comment:
can we also test that the message included here is only the most recent one?

Reply:
Of course, I'll do my best here.